packages/compiler/core/checker.ts

```diff
@@ -1985,17 +1988,23 @@ export function createChecker(program: Program): Checker {
       }
 
       // check "global scope" declarations
-      binding = resolveIdentifierInTable(
+      const globalBinding = resolveIdentifierInTable(
         node,
         globalNamespaceNode.symbol.exports,
         resolveDecorator
       );
 
-      if (binding) return binding;
-
       // check using types
-      binding = resolveIdentifierInTable(node, scope.locals, resolveDecorator);
-      if (binding) return binding.flags & SymbolFlags.DuplicateUsing ? undefined : binding;
+      const usingBinding = resolveIdentifierInTable(node, scope.locals, resolveDecorator);
+
+      if (globalBinding && usingBinding) {
+        reportAmbiguousIdentifier(node, [globalBinding, usingBinding]);
+        return globalBinding;
+      } else if (globalBinding) {
+        return globalBinding;
+      } else if (usingBinding) {
+        return usingBinding.flags & SymbolFlags.DuplicateUsing ? undefined : usingBinding;
+      }
```

Comment:
This is pleasantly simpler than I expected! 😍

Reply:
Yes, me too... though I won't tell you how long it took me to figure out 😆